pkg/manifest/subscription_test.go

```diff
@@ -0,0 +1,425 @@
+								subkeyFilepath,
+							},
+							Wants: serviceWants,
+							After: serviceAfter,
+						},
+						Service: &osbuild.Service{
+							Type: osbuild.OneshotServiceType,
+							ExecStart: []string{
+								"/usr/sbin/subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY} --serverurl theserverurl-etc --baseurl thebaseurl-etc",
+								"/usr/bin/rm " + subkeyFilepath,
+							},
+							EnvironmentFile: []string{
+								subkeyFilepath,
+							},
+						},
+						Install: &osbuild.Install{
+							WantedBy: serviceWantedBy,
+						},
+					},
+				},
+			},
+			expectedFiles: []*fsnode.File{
+				mkKeyfile("theorg-etc", "thekey-etc"),
+			},
+			expectedDirs:     make([]*fsnode.Directory, 0),
+			expectedServices: []string{serviceFilename},
+		},
+		"insights-on-boot+etc-unit-path": {
+			subOpts: subscription.ImageOptions{
+				Organization:  "theorg-iob-etc",
+				ActivationKey: "thekey-iob-etc",
+				ServerUrl:     "theserverurl-iob-etc",
+				BaseUrl:       "thebaseurl-iob-etc",
+				Insights:      true,
+				Rhc:           false,
+			},
+			srvcOpts: &manifest.SubscriptionServiceOptions{
+				InsightsOnBoot: true,
+				UnitPath:       osbuild.EtcUnitPath,
+			},
+			expectedStage: &osbuild.Stage{
+				Type: stageType,
+				Options: &osbuild.SystemdUnitCreateStageOptions{
+					Filename: serviceFilename,
+					UnitType: unitType,
+					UnitPath: osbuild.EtcUnitPath,
+					Config: osbuild.SystemdServiceUnit{
+						Unit: &osbuild.Unit{
+							Description: serviceDescription,
+							ConditionPathExists: []string{
+								subkeyFilepath,
+							},
+							Wants: serviceWants,
+							After: serviceAfter,
+						},
+						Service: &osbuild.Service{
+							Type: osbuild.OneshotServiceType,
+							ExecStart: []string{
+								"/usr/sbin/subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY} --serverurl theserverurl-iob-etc --baseurl thebaseurl-iob-etc",
+								"/usr/bin/insights-client --register", // added when insights is enabled
+								"restorecon -R /root/.gnupg",          // added when insights is enabled
+								"/usr/bin/rm " + subkeyFilepath,
+							},
+							EnvironmentFile: []string{
+								subkeyFilepath,
+							},
+						},
+						Install: &osbuild.Install{
+							WantedBy: serviceWantedBy,
+						},
+					},
+				},
+			},
+			expectedFiles: []*fsnode.File{
+				mkKeyfile("theorg-iob-etc", "thekey-iob-etc"),
+				mkInsightsDropinFile(),
+			},
+			expectedDirs:     []*fsnode.Directory{mkInsightsDropinDir()},
+			expectedServices: []string{serviceFilename},
+		},
+	}
+
+	for name := range testCases {
+		tc := testCases[name]
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			stage, dirs, files, services, err := manifest.SubscriptionService(tc.subOpts, tc.srvcOpts)
+			assert.NoError(err)
+			assert.Equal(stage, tc.expectedStage)
+			assert.Equal(dirs, tc.expectedDirs)
+			assert.Equal(files, tc.expectedFiles)
```

Comment:
It may be good to extend the test to check that none of the files and dirs have `User` and `Group` set to `nil`.

Reply:
Good idea!